js/main.js

```diff
@@ -30,12 +47,17 @@ function handleFormSubmit(event) {
     notes: $form.notes.value,
     entryID: data.nextEntryId
   };
-  data.entries.push(entry);
+  if(data.editing != null) {
+    data.entries[data.editing] = entry;
+    data.editing = null;
```

Comment:
You shouldn't be giving a new id to an edited entry, only new entries get new ids, this should reuse the old id